### ISSUE TITLE
PostalCode/Country existence validation

### ISSUE THREAD

**Issue by ghazispeakserve** [Respect/Validation#285]
Hello,

Is there anyway that we can check if a country has a PostalCode address?

I can see taht $postalCodes variable in PostalCode is not static, we need to create a PostalCode instance to do an 'in_array'.

Any idea about how to resolve this issue?

Regards,
Ghazi

**Comment by henriquemoody** [comment]
I really want to avoid using static stuff.

I have to make a more accurate look but we can add another rule, like `hasPostalCode()` what you think?

**Comment by ghazispeakserve** [comment]
Adding ```->hasPostalCode('TN')->``` is not a bad idea. What I want to achieve is : if a selected country has a postal code then we validate it, else we accept it as it is.

**Comment by henriquemoody** [comment]
You can use `v::when(v::hasPostalCode('TN'), v::postalCode('TN'));`

**Comment by henriquemoody** [comment]
I sent #287 in order to fix some wrong behaviors on `postalCode()` rule:

- It should not be case-sensitive
- It should not throws exception when the country code is valid but doesn't have postal codes
- It should allow only empty strings when country doesn't have postal codes

With these fixes I think creating `hasPostalCode()` is not necessary.

Any thoughts, @ghazispeakserve?

**Comment by ghazispeakserve** [comment]
It seems to be better. What I am doing now is

```php
// create of fake postal code
$pc = new PostalCode('US');
if (in_array($this->data['country'], array_keys($pc->postalCodes)) !== FALSE) {
   V::create()->notEmpty()->postalCode($this->data['country'])->check($this->data['post_code']);
}
```

I think now I can safely use your update without the of the fake instance I create. Is that right?

**Comment by henriquemoody** [comment]
Yeah, I did merge it already and it's available on 0.7.4 version and 0.8.x-dev.

**Comment by henriquemoody** [comment]
Note: You have to remove `notEmpty()` from your chain, since some countries don't have postal code.
